### ISSUE TITLE
Wrong Netplan config causes ax/ac/bg/n wpa cases failed

### ISSUE THREAD

**Issue by baconYao** [canonical/checkbox#1303]
### Bug Description

The wrong netplan config used in Checkbox causes WPA cases failed.

Wrong config like below [1]:
```yaml
# This is the network config written by checkbox
network:
  version: 2
  wifis:
    wlp1s0:
      access-points:
        cert-ac-wpa-tel-l4:
          auth:
            password: insecure
      dhcp4: true
      nameservers: {}
```

Add `key-management: psk` in `auth` can solve this issue.

[1]: Submission: https://certification.canonical.com/hardware/202307-31859/submission/375752/test-results/?term=_np

### To Reproduce

1. Install Checkbox Snap on UC22 image
2. Run `wireless-automated` test plan

### Environment

- OS: UC22
- checkbox22: `4.0.0-dev312    984    latest/beta    ce-certification-qa`

**Comment by baconYao** [comment]
The default configuration of Netplan generated by `console-conf` is like below and it's working

```yaml
# This is the network config written by 'console-conf'
network:
  ethernets:
    eth0:
      dhcp4: true
  version: 2
  wifis:
    wlp1s0:
      access-points:
        cert-ac-wpa-tel-l4:
          password: insecure
      dhcp4: true
```

**Comment by zongminl** [comment]
Will be fixed by #1301

**Comment by p-gentili** [comment]
To add to the above, we went from (270):
```
# This is the network config written by checkbox
network:
  version: 2
  wifis:
    wlan0:
      access-points:
        cert-ac-wpa-tel-l3-01: {password: insecure}
      addresses: []
      dhcp4: True
      nameservers: {}
 ```
 
 to (312):
```
# This is the network config written by checkbox
network:
  version: 2
  wifis:
    wlan0:
      access-points:
        cert-ac-wpa-tel-l3-01:
          auth:
            password: insecure
      dhcp4: true
      nameservers: {}
```

**Comment by Hook25** [comment]
Fixed [dae49f5](https://github.com/canonical/checkbox/commit/dae49f57fe73c8535c2ecb32b85b9d277e12edaa)

**Comment by baconYao** [comment]
Verify Pass on G510 UC22
  - Submission: https://certification.canonical.com/hardware/202405-34023/submission/384472/

```bash
ceqa@localhost:~$ snap list
Name                                    Version         Rev    Tracking       Publisher            Notes
bluez                                   5.64-6          382    22/stable      canonical✓           -
bugit                                   1.2             35     latest/stable  pieq                 devmode
checkbox-baoshan                        0.5dev-jammy    140    latest/edge    ce-certification-qa  devmode
checkbox-ce-oem                         1.0-jammy       365    uc22/edge      ce-certification-qa  devmode
checkbox22                              4.1.0-dev25     1025   latest/beta    ce-certification-qa  -
```
